tests/views/test_describe_image.py

```diff
@@ -0,0 +1,178 @@
+        ),
+    }
+
+
+def test_success(admin_client, settings):
+    settings.WAGTAIL_AI = {
+        "BACKENDS": {
+            "echo": {
+                "CLASS": "wagtail_ai.ai.echo.EchoBackend",
+                "CONFIG": {
+                    "MODEL_ID": "echo",
+                    "TOKEN_LIMIT": 123123,
+                },
+            },
+        },
+        "IMAGE_DESCRIPTION_BACKEND": "echo",
+    }
+
+    image = cast(Image, ImageFactory())
+    response = admin_client.post(
+        reverse("wagtail_ai:describe_image"), data={"image_id": image.pk}
+    )
+    assert response.status_code == 200
+    assert response.json() == {
+        "message": "This is an echo backend: images/example.max-800x600.jpg"
+    }
+
+
+def test_custom_prompt(admin_client, settings, monkeypatch: pytest.MonkeyPatch):
+    describe_image = Mock(return_value=echo.EchoResponse(iter([])))
+    monkeypatch.setattr(echo.EchoBackend, "describe_image", describe_image)
+
+    CUSTOM_PROMPT = "Other prompt"
+    settings.WAGTAIL_AI = {
+        "BACKENDS": {
+            "echo": {
+                "CLASS": "wagtail_ai.ai.echo.EchoBackend",
+                "CONFIG": {
+                    "MODEL_ID": "echo",
+                    "TOKEN_LIMIT": 123123,
+                },
+            },
+        },
+        "IMAGE_DESCRIPTION_BACKEND": "echo",
+        "IMAGE_DESCRIPTION_PROMPT": CUSTOM_PROMPT,
+    }
+
+    image = cast(Image, ImageFactory())
+    admin_client.post(reverse("wagtail_ai:describe_image"), data={"image_id": image.pk})
+    assert describe_image.call_args == call(image_file=ANY, prompt=CUSTOM_PROMPT)
+
+
+def test_custom_rendition_filter(admin_client, settings):
+    settings.WAGTAIL_AI = {
+        "BACKENDS": {
+            "echo": {
+                "CLASS": "wagtail_ai.ai.echo.EchoBackend",
+                "CONFIG": {
+                    "MODEL_ID": "echo",
+                    "TOKEN_LIMIT": 123123,
+                },
+            },
+        },
+        "IMAGE_DESCRIPTION_BACKEND": "echo",
+        "IMAGE_DESCRIPTION_RENDITION_FILTER": "max-100x100",
+    }
+
+    image = cast(Image, ImageFactory())
+    response = admin_client.post(
+        reverse("wagtail_ai:describe_image"), data={"image_id": image.pk}
+    )
+    assert response.status_code == 200
+    assert response.json() == {
+        "message": "This is an echo backend: images/example.max-100x100.jpg"
+    }
+
+
+@pytest.mark.parametrize(
+    "maxlength,expected_status,error_message",
+    [
+        (100, HTTPStatus.OK, None),
+        (
+            -1,
+            HTTPStatus.BAD_REQUEST,
+            "Ensure this value is greater than or equal to 0.",
+        ),
+        (
+            10000,
+            HTTPStatus.BAD_REQUEST,
+            "Ensure this value is less than or equal to 4096.",
+        ),
+    ],
+)
+def test_maxlength_validation(
+    admin_client, settings, maxlength, expected_status, error_message
+):
```

Comment:
**Nitpick:** I've just realised we're missing type signatures on all the tests. I think it would be nice to have that. I am not sure how well it plays with pytest in general though.

Reply:
It works fine, I add them when I want my code editor to be smarter: https://github.com/wagtail/wagtail-ai/pull/81/files#diff-7765e78ff878e5243818245b2482ef977b5920fc88e886022ad958b8c837bc0cR172

But yes, I agree we should add them to all the tests.

Reply:
Added an issue for this #86